python/lsst/display/firefly/footprints.py

```diff
@@ -0,0 +1,202 @@
+#
+# LSST Data Management System
+# Copyright 2018 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+##
+# @file
+# @brief Support for LSST footprints, specific to Firefly
+
+from copy import deepcopy
+import io
+import tempfile
+
+import numpy as np
+from astropy.io.votable.tree import VOTableFile, Resource, Field, Info
+from astropy.io.votable import from_table
+from astropy.table import Table
+import astropy.units as u
+
+import lsst.afw.geom as afwGeom
+
+
+
+def record_selector(record, selection):
+    """Select records from source catalog
+
+    Parameters:
+    -----------
+    record : `lsst.afw.detect.SourceRecord`
+        record to select
+    selection : `str`
+        'all' to select all records. 'blended parents' to select records with
+        more than zero children. 'deblended children' to select records with
+        non-zero parents. 'isolated' to select records that are not blended,
+        meaning zero parents and zero children.
+        Values to check for sel
+    """
+    nChildren = record.get('deblend_nChild')
+    parentId = record.getParent()
+    if selection == 'all':
+        return True
+    elif selection == 'blended parents':
+        return (nChildren > 0)
+    elif selection == 'deblended children':
+        return (parentId > 0)
+    elif selection == 'isolated':
+        return ((parentId == 0) and (nChildren == 0))
+    else:
+        raise RuntimeError('invalid selection: {}'.format(selection) + 
+                           '\nMust be one of "all", "blended parents", ' +
+                           '"deblended children", "isolated"')
+
+
+def create_footprints_table(catalog, xy0 = afwGeom.Point2I(0,0),
+                            insert_column=6):
```

Comment:
`Point2I`, so it's [not safe to use as  default argument](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments).

Also, no spaces around `=` in kwargs, I think; did you run flake8 on this?

Reply:
Thank you for pointing out this gotcha! I've changed the default to be None which then sets it to `Point2I(0, 0)`. Also I discovered I broke my flake8 setup in SublimeText so I've been missing a lot of formatting issues.